File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/2052/BusinessApplication/Helpers/TargetNullValueConverter.vb
Imports System
Imports System.Windows.Data

''' <summary>
''' 双向 IValueConverter，使您可以将可绑定对象的属性(可以为空 string 值)绑定到依赖属性(在该情况下应设置为 null)。
''' </summary>
Public Class TargetNullValueConverter
    Implements IValueConverter

    ''' <summary>
    ''' 将 <c>null</c> 或空 strings 转换为 <c>null</c>。
    ''' </summary>
    ''' <param name="value">要转换的值。</param>
    ''' <param name="targetType">预期结果 type (已忽略)。</param>
    ''' <param name="parameter">可选参数(已忽略)。</param>
    ''' <param name="culture">转换的区域性(已忽略)。</param>
    ''' <returns>如果 <paramref name="value"/> 为 <c>null</c> 或为空，则此方法返回 <c>null</c>，否则返回 <paramref name="value"/>。</returns>
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Dim strValue As String = Nothing

        If TypeOf value Is String Then
            strValue = DirectCast(value, String)
        End If

        Return If(String.IsNullOrEmpty(strValue), Nothing, value)
    End Function

    ''' <summary>
    ''' 将 <c>null</c> 转换回 <see cref="String.Empty"/>。
    ''' </summary>
    ''' <param name="value">要转换的值。</param>
    ''' <param name="targetType">预期结果 type (已忽略)。</param>
    ''' <param name="parameter">可选参数(已忽略)。</param>
    ''' <param name="culture">转换的区域性(已忽略)。</param>
    ''' <returns>如果 <paramref name="value"/> 为 <c>null</c>，则它返回 <see cref="String.Empty"/>，否则返回 <paramref name="value"/>。</returns>
    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return If(value Is Nothing, String.Empty, value)
    End Function
End Class

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1041/BusinessApplication/Helpers/TargetNullValueConverter.cs
namespace $safeprojectname$
{
    using System;
    using System.Windows.Data;

    /// <summary>
    /// 空の string 値を指定できるバインド可能なオブジェクトのプロパティを依存関係プロパティ (その場合、null に設定する必要があります) にバインドできる、双方向の IValueConverter です。
    /// </summary>
    public class TargetNullValueConverter : IValueConverter
    {
        /// <summary>
        /// <c>null</c> または空の strings を <c>null</c> に変換します。
        /// </summary>
        /// <param name="value">変換する値です。</param>
        /// <param name="targetType">結果に必要な型です (無視)。</param>
        /// <param name="parameter">省略可能なパラメーターです (無視)。</param>
        /// <param name="culture">変換のカルチャです (無視)。</param>
        /// <returns><paramref name="value"/> が <c>null</c> または空の場合、このメソッドは <c>null</c> を返し、それ以外の場合は <paramref name="value"/> を返します。</returns>
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            string strValue = value as string;

            return string.IsNullOrEmpty(strValue) ? null : value;
        }

        /// <summary>
        /// <c>null</c> を変換して <see cref="String.Empty"/> に戻します。
        /// </summary>
        /// <param name="value">変換する値です。</param>
        /// <param name="targetType">結果に必要な型です (無視)。</param>
        /// <param name="parameter">省略可能なパラメーターです (無視)。</param>
        /// <param name="culture">変換のカルチャです (無視)。</param>
        /// <returns><paramref name="value"/> が <c>null</c> の場合は <see cref="String.Empty"/> を返し、それ以外の場合は <paramref name="value"/> を返します。</returns>
        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return value ?? string.Empty;
        }
    }
}
